packages/protocol/scripts/build.ts

```diff
@@ -103,14 +105,31 @@ function compile(outdir: string) {
   for (const externalContractPackage of externalContractPackages) {
     console.log(`Building external contracts for ${externalContractPackage.name}`)
     exec(
-      `yarn run truffle compile --silent --contracts_directory=./lib/${externalContractPackage.path}/contracts --contracts_build_directory=./build/contracts-${externalContractPackage.name}`
+      `yarn run truffle compile --silent --contracts_directory=./lib/${externalContractPackage.path}/contracts --contracts_build_directory=./build/contracts-${externalContractPackage.name}` // todo change to outdir
     )
   }
 
+  // compile 0.8 contracts
+  const contracts08 = `./contracts-0.8`
+  console.log(contracts08)
+  if (fs.existsSync(contracts08)) {
+    // fs.mkdirSync(contracts08)
+
+    console.log(`Building contracts using 0.8`)
+    exec(
+      `yarn run truffle compile  --contracts_directory="./contracts-0.8" --contracts_build_directory=${outdir}/contracts-0.8 --config truffle-config0.8.js`
+    )
+  } else {
+    console.log("0.8 folder doesn't exist")
```

Comment:
Should we just create it in an `if (!fs.existsSync())` before this?

Reply:
I don't think so because when we check out old versions it the folder may not exist.